File: manuel/Class 3/집합/set.cpp
#include <iostream>
#include <string>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int M;
    cin >> M;
    int s_mask = 0;

    for(int i = 0; i < M; i++)
    {
        string op;
        cin >> op;

        if(op == "add")
        {
            int x;
            cin >> x;
            s_mask |= (1 << (x - 1));
        }
        else if(op == "remove")
        {
            int x;
            cin >> x;
            s_mask &= ~(1 << (x - 1));
        }
        else if(op == "check")
        {
            int x;
            cin >> x;
            if((s_mask & (1 << (x - 1))) != 0)
            {
                cout << 1 << "\n";
            }
            else
            {
                cout << 0 << "\n";
            }
        }
        else if(op == "toggle")
        {
            int x;
            cin >> x;
            s_mask ^= (1 << (x - 1));
        }
        else if(op == "all")
        {
            s_mask = (1 << 20) - 1;
        }
        else if(op == "empty")
        {
            s_mask = 0;
        }
    }

    return 0;
}

File: manuel/Class 3/집합/set.py
import sys

def solve():
    M = int(sys.stdin.readline())
    s_mask = 0

    for _ in range(M):
        command = sys.stdin.readline().split()
        op = command[0]

        if op == "add":
            x = int(command[1])
            s_mask |= (1 << (x - 1))
        elif op == "remove":
            x = int(command[1])
            s_mask &= ~(1 << (x - 1))
        elif op == "check":
            x = int(command[1])
            if(s_mask & (1 << (x - 1))) != 0:
                print(1)
            else:
                print(0)
        elif op == "toggle":
            x = int(command[1])
            s_mask ^= (1 << (x - 1))
        elif op == "all":
            s_mask = (1 << 20) - 1
        elif op == "empty":
            s_mask = 0

solve()